components/search/src/search.udl

```diff
@@ -10,8 +10,200 @@ interface SearchApiError {
+};
+
+/**
+ * A definition for an individual search engine to be presented to the user.
+ */
+dictionary SearchEngineDefinition {
+    /// An optional list of aliases for this engine.
+    sequence<string>? aliases;
+
+    /// The classification of search engine according to the main search types
+    /// (e.g. general, shopping, travel, dictionary). Currently, only marking as
+    /// a general search engine is supported.
+    /// On Android, only general search engines may be selected as "default"
+    /// search engines.
+    string classification;
+
+    /// The identifier of the search engine. This is used as an internal
+    /// identifier, e.g. for saving the user's settings for the engine. It is
+    /// also used to form the base telemetry id and may be extended by telemetrySuffix.
+    string identifier;
+
+    /// The user visible name of the search engine.
+    string name;
+
+    /// The partner code for the engine. This will be inserted into parameters
+    /// which include `{partnerCode}`.
+    string? partner_code;
+
+    /// Optional suffix that is appended to the search engine identifier
+    /// following a dash, i.e. `<identifier>-<suffix>`
+    string? telemetry_suffix;
+
+    /// The URLs associated with the search engine.
+    SearchEngineUrls urls;
+};
+
+/**
+ * Details of the search engines to display to the user, generated as a result
+ * of processing the search configuration.
+ */
+dictionary FilteredSearchEngines {
+    /// A list of engines, with the default engine first, and the rest in the
+    /// order defined by the configuration.
+    sequence<SearchEngineDefinition> engines;
+
+    /// The identifier of the engine that should be used for the application
+    /// default engine.
+    string app_default_engine_id
+
+    /// If specified, the identifier of the engine that should be used for the
+    /// application default engine in private browsing mode.
+    /// Only desktop uses this currently.
+    string? app_default_private_engine_id;
+};
+
+/**
+ * The possible application names.
+ */
+enum SearchApplicationName {
+    "firefox",
+    "firefox-android",
+    "focus-android",
+    "firefox-ios",
+    "focus-ios",
+};
+
+/**
+ * The possible channels that may be in use.
+ */
+enum SearchDistributionChannel {
+    "default",
+    "nightly",
+    "aurora",
+    "beta",
+    "release",
+    "esr",
+};
+
+/**
+ * The user's environment that is used for filtering the search configuration.
+ */
+dictionary SearchUserEnvironment {
+    /// The current locale of the application that the user is using.
+    string locale;
+
+    /// The home region that the user is currently identified as being within.
+    /// On desktop & android there is a 14 day lag after detecting a region
+    /// change before the home region changes. TBD: iOS?
+    string region;
+
+    /// The current distribution channel.
+    /// Use `default` for a self-build or an unknown channel.
+    SearchDistributionChannel channel;
```

Comment:
Will this allow us to support different search engine list for distribution partnerships? Or should we use distribution_id below?

example here is a manufacturer that preloads/preinstalls Fenix in their Android OS but want to add a search engine (default or not) as a part of the partnership?